maven-plugin-tools-generators/src/main/java/org/apache/maven/tools/plugin/generator/GeneratorUtils.java

```diff
@@ -122,14 +122,19 @@ public static void element( XMLWriter w, String name, String value, boolean asTe
     
     /**
      * @param artifacts not null collection of <code>Artifact</code>
-     * @return list of component dependencies
+     * @return list of component dependencies, only in compile scope
      */
     public static List<ComponentDependency> toComponentDependencies( Collection<Artifact> artifacts )
     {
         List<ComponentDependency> componentDeps = new LinkedList<>();
 
         for ( Artifact artifact : artifacts )
         {
+            if ( !Artifact.SCOPE_COMPILE.equals( artifact.getScope() ) )
```

Comment:
I would rather make `SCOPE = PROVIDED` (then continue). What about runtime scope? Just to be on safe side.

Reply:
fixed